src/devices/machine/upd765.cpp

```diff
@@ -719,13 +761,18 @@ void upd765_family_device::fifo_expect(int size, bool write)
 
 uint8_t upd765_family_device::dma_r()
 {
+	uint8_t data = fifo_pop(false);
+
+	LOG("%s d:%02X\n", FUNCNAME, data);
+
 	if(machine().side_effects_disabled())
 		return fifo[0];
-	return fifo_pop(false);
+	return data;
```

Comment:
You’ve broken this – it’s no longer suppressing the FIFO pop if side effects are disabled.